slick/src/main/scala/slick/jdbc/DriverDataSource.scala

```diff
@@ -46,42 +46,45 @@ class DriverDataSource(
   // State that gets initialized by `init`
   @volatile private[this] var registered: Boolean = false
   @volatile private[this] var initialized = false
-  private[this] var driver: Driver = _
-  private[this] var connectionProps: Properties = _
+  @volatile private[this] var driver: Driver = _
+  @volatile private[this] var connectionProps: Properties = _
```

Comment:
Ahh. You are right. That section had fallen out of my memory.

From: https://www.cs.umd.edu/~pugh/java/memoryModel/jsr-133-faq.html#volatile
"In effect, because the new memory model places stricter constraints on reordering of volatile field accesses with other field accesses, volatile or not, anything that was visible to thread A when it writes to volatile field f becomes visible to thread B when it reads f."

Thanks for the refresh.

Will add an `init` to `getParentLogger` and update the PR later tonight.